app/graph/mutations/ocr_mutations.rb

```diff
@@ -1,15 +1,16 @@
 module OcrMutations
-  ExtractText = GraphQL::Relay::Mutation.define do
-    name 'ExtractText'
+  class ExtractText < Mutations::BaseMutation
+    argument :id, ID, required: true
 
-    input_field :id, !types.ID
+    field :project_media, ProjectMediaType, null: true, camelize: false
 
-    return_field :project_media, ProjectMediaType
-
-    resolve -> (_root, inputs, ctx) {
-      pm = GraphqlCrudOperations.object_from_id_if_can(inputs['id'], ctx['ability'])
+    def resolve(**inputs)
+      pm = GraphqlCrudOperations.object_from_id_if_can(
+        inputs[:id],
+        context[:ability]
```

Comment:
yep! object and context are always available in resolve functions. also, the arguments are actually passed as keyword arguments, so going forward we should just use them in the method signature when known rather than pulling it via a hash key inside the function.

here's more documentation on that: https://graphql-ruby.org/fields/resolvers.html#when-do-you-really-need-a-resolver

that same article recommends moving toward having plain ol ruby functions (that you can unit test!) rather than using resolving functions where possible, which is a suggestion I liked. here's the pull quote:

> Do you really need a Resolver? Putting logic in a Resolver has some downsides:
> Since it’s coupled to GraphQL, it’s harder to test than a plain ol’ Ruby object in your app
> Since the base class comes from GraphQL-Ruby, it’s subject to upstream changes which may require updates in your code
> Here are a few alternatives to consider:
> * Put display logic (sorting, filtering, etc.) into a plain ol’ Ruby class in your app, and test that class
> * Hook up that object with a method, for example:
> ```
> field :recommended_items, [Types::Item], null: false
> def recommended_items
> ItemRecommendation.new(user: context[:viewer]).items
> end
> ```